src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -394,6 +396,11 @@ public function load(array $configs, ContainerBuilder $container)
             // remove tagged iterator argument for resource checkers
             $container->getDefinition('config_cache_factory')->setArguments([]);
         }
+
+        if ($config['disallow_search_engine_index'] ?? false) {
+            $container->register(DisallowRobotsIndexingListener::class)
```

Comment:
The service should be part of `debug.xml` file instead.

Reply:
What if the configuration option is activated in prod (like a customer portal as suggested by linaori)?